packages/pwa-buildpack/lib/cli/generate-build-report.js

```diff
@@ -101,19 +101,10 @@ function getYarnDependencies(dependencies) {
 
 function getNpmDependencies(dependencies) {
     const dependencyMap = new Map();
+    const lockFile = require(path.resolve(cwd, 'package-lock.json'));
 
     dependencies.map(packageName => {
-        const listbuffer = spawnSync('npm', ['list', packageName]);
-        const grepBuffer = spawnSync('grep', ['└──'], {
-            input: listbuffer.stdout
-        });
-
-        // TODO: handle multiple versions installed when not dupe. ie "npm ls core-js"
-        const [, , version] = grepBuffer.stdout
-            .toString()
-            .split('\n') // ["│ └── @magento/upward-js@5.0.0  deduped", "└── @magento/upward-js@5.0.0"]
-            .filter(text => !text.includes('deduped'))[0] // ["└── @magento/upward-js@5.0.0"]
-            .split('@'); // [ '└── ', 'magento/upward-js', '5.0.0' ]
+        const version = lockFile.dependencies[packageName].version;
```

Comment:
Ha this is so much simpler. Does this handle multiple versions of a dependency such as `core-js`? I know we don't necessarily need to worry about that since we're just checking Magento packages right now, but it was something I didn't figure out before with the TODO.

Reply:
Nice question dude. I did check for that and added a loop construct in the YARN version of this function to handle multiple versions. Coming to NPM it is handled a little differently. package-lock.json is only holding on to 1 version of the package.

Reply:
> package-lock.json is only holding on to 1 version of the package.

Are you sure? If you do `npm ls core-js` it'll show two resolved versions.

Reply:
@revanth0212 this is from a scaffolded app running `npm ls core-js`.

```
~/code/v9test
λ npm ls core-js
v9test@0.0.1 /Users/stephen/code/v9test
├─┬ @storybook/react@5.2.8
│ ├─┬ @storybook/addons@5.2.8
│ │ ├─┬ @storybook/api@5.2.8
│ │ │ └── core-js@3.8.2  deduped
│ │ ├─┬ @storybook/channels@5.2.8
│ │ │ └── core-js@3.8.2  deduped
│ │ ├─┬ @storybook/client-logger@5.2.8
│ │ │ └── core-js@3.8.2  deduped
│ │ ├─┬ @storybook/core-events@5.2.8
│ │ │ └── core-js@3.8.2  deduped
│ │ └── core-js@3.8.2  deduped
│ ├─┬ @storybook/core@5.2.8
│ │ ├─┬ @storybook/channel-postmessage@5.2.8
│ │ │ └── core-js@3.8.2  deduped
│ │ ├─┬ @storybook/client-api@5.2.8
│ │ │ └── core-js@3.8.2  deduped
│ │ ├─┬ @storybook/router@5.2.8
│ │ │ └── core-js@3.8.2  deduped
│ │ ├─┬ @storybook/theming@5.2.8
│ │ │ └── core-js@3.8.2  deduped
│ │ ├─┬ @storybook/ui@5.2.8
│ │ │ ├─┬ @storybook/components@5.2.8
│ │ │ │ ├── core-js@3.8.2  deduped
│ │ │ │ └─┬ simplebar-react@1.2.3
│ │ │ │   └─┬ simplebar@4.2.3
│ │ │ │     └── core-js@3.8.2  deduped
│ │ │ └── core-js@3.8.2  deduped
│ │ ├── core-js@3.8.2  deduped
│ │ └─┬ lazy-universal-dotenv@3.0.1
│ │   └── core-js@3.8.2  deduped
│ ├─┬ @storybook/node-logger@5.2.8
│ │ └── core-js@3.8.2  deduped
│ └── core-js@3.8.2
└─┬ graphql-cli@3.0.14
  ├─┬ apollo-codegen@0.20.2
  │ ├─┬ apollo-codegen-core@0.20.1
  │ │ └── core-js@2.6.12
  │ └─┬ apollo-codegen-flow@0.20.0
  │   └─┬ apollo-codegen-core@0.20.1
  │     └── core-js@2.6.12
  └─┬ tmp-graphql-config-extension-openapi@1.0.7
    └─┬ @kbrandwijk/swagger-to-graphql@2.4.3
      ├─┬ babel-runtime@6.26.0
      │ └── core-js@2.6.12
      └─┬ json-schema-ref-parser@3.3.1
        └─┬ z-schema@3.25.1
          └── core-js@2.6.12
```

Reply:
You are right about `npm ls`. But there is a small caviate, It is listing all the packages that `require` the package, not the list of installed versions.

For instance:
![image](https://user-images.githubusercontent.com/35203638/103915901-694d6980-50d1-11eb-8c86-73cfb74155d5.png)

At the moment there is only 1 version of it present in the `lock` file.
![image](https://user-images.githubusercontent.com/35203638/103916258-d2cd7800-50d1-11eb-87f2-e505553754ad.png)

On the other hand, there can be packages that require a diff version of the package, for those, the package will be listed in the dependencies section of those packages.
![image](https://user-images.githubusercontent.com/35203638/103916450-0f00d880-50d2-11eb-9737-74f62929d9ce.png)

After combing through the lock file for `core-js` I realized we have to look for all the packages that have our package as a dependency, which means we have to dig deeper than we are doing at the moment. I will work on that push the changes in the next commits.

Thanks for pointing this out. Good work buddy.